support-data-rest/src/main/java/uk/ac/ebi/uniprot/api/disease/DiseaseController.java

```diff
@@ -3,47 +3,72 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.ApplicationEventPublisher;
 import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
 import org.springframework.validation.annotation.Validated;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 import uk.ac.ebi.uniprot.api.common.repository.search.QueryResult;
-import uk.ac.ebi.uniprot.api.rest.pagination.PaginatedResultsEvent;
+import uk.ac.ebi.uniprot.api.rest.controller.BasicSearchController;
+import uk.ac.ebi.uniprot.api.rest.output.context.MessageConverterContext;
+import uk.ac.ebi.uniprot.api.rest.output.context.MessageConverterContextFactory;
+import uk.ac.ebi.uniprot.api.rest.validation.ValidReturnFields;
 import uk.ac.ebi.uniprot.cv.disease.Disease;
+import uk.ac.ebi.uniprot.search.field.DiseaseField;
 
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import javax.validation.Valid;
 import javax.validation.constraints.Pattern;
+import java.util.Optional;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static uk.ac.ebi.uniprot.api.rest.output.UniProtMediaType.*;
 
 @RestController
 @RequestMapping("/disease")
 @Validated
-public class DiseaseController {
+public class DiseaseController extends BasicSearchController<Disease> {
```

Comment:
I did not find the Download method, will you implement it as part of another task?

Reply:
There is a separate ticket for download.